Add process package that enables control of external processes

lispmeister opened the pull request:
This package gives you control over an external process. You
can spawn any process and have control over STDIN, STDOUT and
STDERR of the spawned process. All communications is done via
async IO and will not block. Any errors during operation related to
system calls are surfaced to the caller as ProcessError via the
notifier.

- [X] Fix tests when building via "make test"

jemc commented on packages/process/process-monitor.pony:
```diff
@@ -0,0 +1,382 @@
+      compile_error "unsupported platform"
+    end
+
+    
```
Style-wise, there should probably be just one empty line between methods declared in the same type.

Also, I notice there are many empty lines at the end of this file - there should be only one (which GitHub will trim in the display because it's conventional to have on every file).

jemc commented on packages/process/foobar:
```diff
@@ -0,0 +1 @@
+the quick brown fox jumps over the lazy dog
```
Especially with the current paradigm of having package tests in the same package, it would be nice if we had a way to test this package without random test fixtures hanging out in the standard library files.

jemc commented:
It would be nice if this could eventually be made to work with windows as well, probably using the `CreateProcess` function.  I understand if you don't have the desire or skillset to do windows development (I don't either), but I think we need to make sure that the interface of this package can work in a cross-platform way.  In my mind, that particularly calls into question some of the posix-specific error primitives.

lispmeister commented:
- Fixed formatting (removed extra lines)
- Fixed _TestWc (no longer needs file foobar)
- Fixed test.pony (we now have correct semantics in case data arrives in chunks)

jemc commented on packages/process/test.pony:
```diff
@@ -0,0 +1,159 @@
+
+  fun apply(h: TestHelper) =>
+    // we expect wc to give a count of
+    // 1 line, 9 words, 44 characters, filename foobar
```
This comment still refers to the contents of the old `foobar` file.

jemc commented on packages/process/test.pony:
```diff
@@ -0,0 +1,159 @@
+  fun timed_out(h: TestHelper) =>
+    h.complete(false)
+
+class ProcessClient is ProcessNotify
```
Since the tests are part of the package, helper classes like this should be made private (with a leading underscore) to prevent cluttering the namespace of the types provided by a package, and to preventusers of the package from trying to use them directly.

lispmeister commented:
Tests are not stable. 
I'll investigate and open a new PR once they are fixed.
